SocketRocket/SRWebSocket.m

```diff
@@ -1378,90 +1417,98 @@ - (void)stream:(NSStream *)aStream handleEvent:(NSStreamEvent)eventCode;
     }
 
     dispatch_async(_workQueue, ^{
-        switch (eventCode) {
-            case NSStreamEventOpenCompleted: {
-                SRFastLog(@"NSStreamEventOpenCompleted %@", aStream);
-                if (self.readyState >= SR_CLOSING) {
-                    return;
-                }
-                assert(_readBuffer);
-                
-                if (self.readyState == SR_CONNECTING && aStream == _inputStream) {
-                    [self didConnect];
-                }
-                [self _pumpWriting];
-                [self _pumpScanner];
-                break;
+        [weakSelf safeHandleEvent:eventCode stream:aStream];
+    });
+}
+
+- (void)safeHandleEvent:(NSStreamEvent)eventCode stream:(NSStream *)aStream
+{
+    __strong typeof(self) strongSelf = self;
```

Comment:
Correct me if I'm wrong, but are you confident this does what you're intending? I believe in the context of a method invocation like this, `self` must be live (otherwise it would have short-circuited at `objc_msgSend(nil)`). Once in the context of an invocation like this, your reference to `self` would already be strong, and simply re-declaring it `__strong` explicitly won't actually do anything.